udf/event_code_points_to_string/udf.sql

```diff
@@ -0,0 +1,43 @@
+CREATE OR REPLACE FUNCTION udf.event_code_points_to_string(code_points ANY TYPE) AS (
+  CODE_POINTS_TO_STRING(
+    IF(
+      code_points IS NULL,
+      NULL,
+      ARRAY(
+        SELECT
+          CASE
+          WHEN
+            i IS NULL
+          THEN
+            NULL
+            -- Skip commas
+          WHEN
+            i >= 43
+          THEN
+            i + 2
+            -- Skip double quote
+          WHEN
+            i >= 34
+          THEN
+            i + 1
+          ELSE
+            i
+          END
+        FROM
+          UNNEST(code_points) AS i
+      )
+    )
+  )
+);
```

Comment:
I'm wondering if there is a good way to handle invalid code points, because certain code points like `udf.event_code_points_to_string([55296])` or ones that are too large, like `udf.event_code_points_to_string([2000000])` will throw an error. There is also `SAFE.CODE_POINTS_TO_STRING` which will return `NULL` instead of an error but it feels like these code points would require special handling.

Reply:
The 55296 code point failure is surprising to me. In general I wanted this to hard fail on too-large (e.g. 2M) because then the entire job will fail and we will have to go in and manually fix things. That might mean dropping some event property or whatnot - eventually we will automate this to stop collecting new values once we are too large.